### PR TITLE
Rename action from `UpdateValidator` to `UpdateValidatorMetadata`

### PR DESCRIPTION
**Pull request by siy**
Changes summary:
- Renamed action from `UpdateValidator` to `UpdateValidatorMetadata`, updated accordingly related classes/methods
- Fixed validator unregistration handling while collecting validator info

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/action/UpdateValidatorMetadataAction.java [hunk 1]
```diff
@@ -23,12 +23,12 @@
...
+class UpdateValidatorMetadataAction implements TransactionAction {
 	private final ECPublicKey validatorKey;
 	private final String name;
 	private final String url;
 
-	UpdateValidatorAction(ECPublicKey validatorKey, String name, String url) {
+	UpdateValidatorMetadataAction(ECPublicKey validatorKey, String name, String url) {
 		this.validatorKey = validatorKey;
 		this.name = name;
 		this.url = url;
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/action/UpdateValidatorMetadataAction.java:34]: We actually have a few more metadata attributes.

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/FunctionalUtils.java [hunk 2]
```diff
@@ -97,4 +101,58 @@
...
+
+	/**
+	 * Return copy of the input set with specified element removed.
+	 *
+	 * @param element element to remove
+	 * @param input input set
+	 *
+	 * @return new set with specified element removed
+	 */
+	static <T> Set<T> removeElement(T element, Set<T> input) {
```
Comment (LukasGasior1) [inline radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/FunctionalUtils.java:113]: nice addition :)

**Review by LukasGasior1 — APPROVED** [review]

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/FunctionalUtils.java (hunk 2 above)
Comment (siy) [reply radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/FunctionalUtils.java:113]: yeah, tired repeating same code again :)

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/action/UpdateValidatorMetadataAction.java (hunk 1 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/action/UpdateValidatorMetadataAction.java:34]: The `UpdateValidatorMetadataAction` is converted into `UpdateValidatorMetadata` instance while building the transaction, and it lists only these two fields.

Comment (iamyulong) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/action/UpdateValidatorMetadataAction.java:34]: Never mind. I confused myself by ValidatorMetadata and TokenResourceMetadata.

Looks good to me!

**Review by iamyulong — APPROVED** [review]

**Review by michalzajda — APPROVED** [review]

**Pull request merged**